src/reporter/tests/test_entities_with_odd_chars.py

```diff
@@ -0,0 +1,81 @@
+from conftest import QL_URL
+import pytest
+import requests
+import time
+import urllib
+from .utils import send_notifications
+
+
+def mk_entity(eid, entity_type, attr_name):
+    return {
+        'id': eid,
+        'type': entity_type,
+        attr_name: {
+            'type': 'Text',
+            'value': 'test'
+        }
+    }
+
+
+def insert_entity(entity):
+    notification_data = [{'data': [entity]}]
+    send_notifications(notification_data)
+
+    time.sleep(2)
+
+
+def query_entity(entity_id, attr_name):
+    escaped_attr_name = urllib.parse.quote(attr_name)
+    url = "{}/entities/{}/attrs/{}".format(QL_URL, entity_id, escaped_attr_name)
+    response = requests.get(url)
+    assert response.status_code == 200
+    return response.json().get('data', {})
+
+
+def delete_entities(entity_type):
+    delete_url = "{}/types/{}".format(QL_URL, entity_type)
+    response = requests.delete(delete_url)
+    assert response.ok
+
+
+def run_test(entity_type, attr_name):
+    entity = mk_entity('d1', entity_type, attr_name)
+
+    insert_entity(entity)
+
+    query_result = query_entity(entity['id'], attr_name)
+    query_result.pop('index', None)
+    assert query_result == {
+        'attrName': attr_name,
+        'entityId': entity['id'],
+        'values': [entity[attr_name]['value']]
+    }
+
+    delete_entities(entity['type'])
```

Comment:
This will open the door for future headaches when running tests because when a test fails, delete will not execute and state will remain in the db.

Have a look at how I did with pytest fixtures for example [here](https://github.com/smartsdk/ngsi-timeseries-api/blob/bc3ca14bdacdc519f7d88706c42aec69b2213d91/src/reporter/tests/test_1T1E1A.py#L27). The two most used fixtures are the "translator" fixture (which cleans up cratedb after the test) and the clean_mongo, both defined in [this file](https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/conftest.py)

Reply:
excellent advice! done it! see latest commit: 855ad50